rootio/scanner.go

```diff
@@ -72,16 +77,17 @@ func (s *baseScanner) Next() bool {
 
 	if s.chain {
 		if s.cur >= s.tot {
-			s.nextTree()
+			ch := s.tree.(*tchain)
+			s.nextTree(ch.cur + 1)
 		}
 	}
 
 	return next
 }
 
-func (s *baseScanner) nextTree() {
+func (s *baseScanner) nextTree(i int) {
```

Comment:
this method should probably be renamed `loadTree`.
it made sense to name it `nextTree` when it wasn't taking any argument, but now this is a bit confusing:
```go
scan.nextTree(3)
```
does this mean we go to the next 3 trees from the current position or the 3rd tree (_ie:_ absolute position) ?
:)

IMHO, `loadTree(i int)` reads better.